benchmarks/distributed/rpc/parameter_server/launcher.py

```diff
@@ -0,0 +1,320 @@
+        trainer_name = get_name(
+            trainer_rank,
+            configurations.trainer_count,
+            configurations.parameter_server_count
+        )
+        ps_rref = None
+        if parameter_server_rrefs:
+            ps_rank = get_parameter_server_rank(trainer_rank, configurations)
+            ps_rref = parameter_server_rrefs[ps_rank]
+        fut = rpc.rpc_async(
+            trainer_name,
+            run_trainer,
+            args=(
+                configurations,
+                trainer_configurations,
+                copy.deepcopy(model),
+                train_data[trainer_rank],
+                trainer_rank,
+                ps_rref,
+            ),
+            timeout=configurations.rpc_async_timeout
+        )
+        futs.append(fut)
+    return futs
+
+
+def benchmark_warmup(
+    configurations, trainer_configurations, ps_configurations, model, data, parameter_server_rrefs
+):
+    if configurations.parameter_server_count > 0:
+        ps = get_benchmark_ps_map()[str(ps_configurations["ps_class"])]
+    futs = call_trainers(configurations, trainer_configurations, model, data, parameter_server_rrefs)
+    for fut in futs:
+        fut.wait()
+    for ps_rref in parameter_server_rrefs.values():
+        rpc.rpc_sync(
+            ps_rref.owner(),
+            ps.reset_state,
+            args=(ps_rref,)
+        )
+    print("benchmark warmup done\n")
+
+
+def split_list(arr, n):
+    return [arr[i::n] for i in range(n)]
+
+
+def run_master(rank, model, data, configurations, trainer_configurations, ps_configurations, rpc_backend_options):
+    world_size = configurations.trainer_count + configurations.parameter_server_count + 1
+    rpc.init_rpc(
+        get_name(
+            rank,
+            configurations.trainer_count,
+            configurations.parameter_server_count
+        ),
+        rank=rank,
+        world_size=world_size,
+        rpc_backend_options=rpc_backend_options
+    )
+    parameter_server_rrefs = {}
+    for i in range(
+        configurations.trainer_count, world_size - 1
+    ):
+        parameter_server_rrefs[i] = get_ps_rref(i, configurations, ps_configurations)
+
+    train_data = split_list(
+        list(DataLoader(data, batch_size=configurations.batch_size)),
+        configurations.trainer_count
+    )
+
+    # warmup run the benchmark
+    benchmark_warmup(
+        configurations, trainer_configurations, ps_configurations, model, train_data, parameter_server_rrefs
+    )
+    # run the benchmark
+    trainer_futs = call_trainers(
+        configurations, trainer_configurations, model, train_data, parameter_server_rrefs
+    )
+    # collect metrics and print
+    metrics_printer = ProcessedMetricsPrinter()
+    rank_metrics_list = [fut.wait() for fut in trainer_futs]
+    metrics_printer.print_metrics("trainer", rank_metrics_list)
+
+
+def run_benchmark(rank, model, data, configurations, trainer_configurations, ps_configurations):
```

Comment:
curious, any reason that configurations, trainer_configurations, ps_configurations needs to be three different objects. The files can be separate, but can they all load into one `config` object and then functions using that config can just retrieve whatever they need?